test/test.js

```diff
@@ -5,111 +5,598 @@ suite('GaiaHeader', function() {
+
+      afterNext(el, 'runFontFit').catch(() => {
+        sinon.assert.notCalled(this.fontFit);
+      }).then(done, done);
+    });
+
+    test('It works when we change `textContent` right after font-fit has been called', function(done) {
+      this.dom.innerHTML = `<gaia-header action="back">
+        <h1>short</h1></gaia-header>`;
+
+      var el = this.dom.firstElementChild;
+      var h1 = el.querySelector('h1');
+
+      h1.textContent = 'long long long long long long long long long long long long long long';
+
+      afterNext(el, 'runFontFit').then(() => {
+        assert.equal(h1.style.fontSize, '16px');
+
+        h1.textContent = 'short';
+
+        return afterNext(el, 'runFontFit').then(() => {
+          assert.equal(h1.style.fontSize, '24px');
+        });
+      }).then(done, done);
+    });
+
+    test('It\'s debounced', function(done) {
+      this.dom.innerHTML = `<gaia-header action="back">
+        <h1>Title</h1>
+      </gaia-header>`;
+
+      var el = this.dom.firstElementChild;
+      var h1 = el.querySelector('h1');
+
+      el.runFontFit();
+      el.runFontFit();
+      el.runFontFit();
+
+      el.runFontFit().then(() => {
+        sinon.assert.calledOnce(el.setTitleStyle);
+      }).then(done, done);
+    });
+
+    test('It always runs when mutation occurs', function(done) {
+      this.dom.innerHTML = `<gaia-header action="back">
+        <h1>Title</h1>
+      </gaia-header>`;
+
+      var el = this.dom.firstElementChild;
+      var h1 = el.querySelector('h1');
+
+      afterNext(el, 'runFontFit').then(() => {
+        return el.runFontFit().then(() => {
+          sinon.assert.calledOnce(el.setTitleStyle);
+          assert.equal(h1.style.fontSize, '16px');
+        });
+      }).then(done, done);
+
+      h1.textContent = 'long long long long long long long long long long long long long long';
+    });
+  });
+
+  suite('Mutation observer', function() {
+    test('It runs font-fit and centering when the title changes', function(done) {
+      this.dom.innerHTML = '<gaia-header action="back"><h1>foo</h1></gaia-header>';
+
+      var el = this.dom.firstElementChild;
+      var h1 = el.querySelector('h1');
+
+      afterNext(el, 'runFontFit').then(() => {
+        this.fontFit.reset();
+        h1.textContent = 'bar';
+
+        return afterNext(el, 'runFontFit').then(() => {
+          sinon.assert.calledOnce(this.fontFit);
+        });
+      }).then(done, done);
+    });
+
+    test('It runs font-fit when `action` changes', function(done) {
+      this.dom.innerHTML = '<gaia-header action="back"><h1>foo</h1></gaia-header>';
+      var el = this.dom.firstElementChild;
+
+      afterNext(el, 'runFontFit').then(() => {
+        this.fontFit.reset();
+
+        // Removing the action should
+        // hide the action-button, so
+        // font-fit needs to be run.
+        el.action = null;
+
+        return afterNext(el, 'runFontFit').then(() => {
+          sinon.assert.calledOnce(this.fontFit);
+        });
```

Comment:
nit: you can refactor this like this:
```javascript
afterNext(el, 'runFontFit').then(() => {
  this.fontFit.reset();

  // Removing the action should
  // hide the action-button, so
  // font-fit needs to be run.
  el.action = null;

  return afterNext(el, 'runFontFit');
}).then(() => {
  sinon.assert.calledOnce(this.fontFit);
}).then(done, done);
```

This is somewhat more readable IMO, and easier to add more assertions if we need it later.